controller/appcontroller.go

```diff
@@ -533,6 +536,12 @@ func (ctrl *ApplicationController) processRequestedAppOperation(app *appv1.Appli
 		// if we just completed an operation, force a refresh so that UI will report up-to-date
 		// sync/health information
 		ctrl.requestAppRefresh(app.ObjectMeta.Name, CompareWithLatest)
+		if key, err := cache.MetaNamespaceKeyFunc(app); err == nil {
```

Comment:
what does this do?

Reply:
The `requestAppRefresh` only forces app comparison during reconciliation loop, but does not trigger the reconciliation loop itself. The `if key, err := cache.MetaNamespaceKeyFunc(app); err == nil { ....etc ` triggers reconciliation loop

Reply:
I think you should comment this behaviour

Reply:
done